File: src/test/java/no/nav/syfo/localconfig/LocalApplicationConfig.java
package no.nav.syfo.localconfig;

import no.nav.security.oidc.test.support.spring.TokenGeneratorConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import org.springframework.core.env.Environment;

import static java.util.Objects.requireNonNull;

@Configuration
@Import(TokenGeneratorConfiguration.class)
public class LocalApplicationConfig {

    public LocalApplicationConfig(Environment environment) {
        System.setProperty("SECURITYTOKENSERVICE_URL", "q12123");
        System.setProperty("SRV_USERNAME", requireNonNull(environment.getProperty("srv.username")));
        System.setProperty("SRV_PASSWORD", requireNonNull(environment.getProperty("srv.password")));
    }
}

File: src/test/kotlin/no/nav/syfo/LocalApplicationConfig.kt
package no.nav.syfo

import no.nav.security.token.support.test.spring.TokenGeneratorConfiguration
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import org.springframework.core.env.Environment
import java.util.*

@Configuration
@Import(TokenGeneratorConfiguration::class)
class LocalApplicationConfig(environment: Environment) {
    init {
        System.setProperty("SECURITYTOKENSERVICE_URL", "q12123")
        System.setProperty("SRV_USERNAME", Objects.requireNonNull(environment.getProperty("srv.username")))
        System.setProperty("SRV_PASSWORD", Objects.requireNonNull(environment.getProperty("srv.password")))
    }
}
